x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termenum/action/TermEnumAction.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.termenum.action;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.query.AbstractQueryBuilder.parseInnerQueryBuilder;
+
+public class TermEnumAction extends ActionType<TermEnumResponse> {
+
+    public static final TermEnumAction INSTANCE = new TermEnumAction();
+//    public static final String NAME = "indices:admin/termsenum/list";
+    public static final String NAME = "indices:data/read/xpack/termsenum/list";
```

Comment:
May be too early to talk about this, but when security is enabled, who can access to this endpoint (those who can read the index))?  what about when FLS and DLS is enabled?

Reply:
DLS and FLS is not part of this PR currently. A TBD.

Reply:
@markharwood Thanks for the answers, Mark. Is this PR is ready for a more thorough review? or you wanted to add more/ do more modifications?

Reply:
Thanks for the review Mayya! There's still some stuff I want to add just yet. (Sorry, I should have added the WIP label).
I need to add logic and tests to filter slower indices based on tiers (we ignore frozen indices and searchable snapshots).

I'll leave DLS and FLS for a subsequent PR.

Reply:
Should it be `indices:data/read/terms` ? No need to add xpack reference here.